sdk/aqueduct/utils.py

```diff
@@ -223,17 +228,24 @@ def _package_files_and_requirements(
         if not os.path.exists(dstfolder):
             os.makedirs(dstfolder)
         shutil.copy(file_path, os.path.join(dir_path, file_path))
-    if os.path.exists(REQUIREMENTS_FILE):
-        Logger.logger.info(
-            "%s: requirements.txt file detected in current directory %s, will not self-generate by inferring package dependencies."
-            % (os.getcwd(), func.__name__)
-        )
-        shutil.copy(REQUIREMENTS_FILE, os.path.join(dir_path, REQUIREMENTS_FILE))
+    if reqs_path:
+        if os.path.exists(reqs_path):
+            Logger.logger.info("Installing requirements found at {path}".format(path=reqs_path))
+            shutil.copy(reqs_path, os.path.join(dir_path, REQUIREMENTS_FILE))
+        else:
+            raise FileNotFoundError("Requirement file provided does not exist.")
     else:
-        Logger.logger.info(
-            "%s: No requirements.txt file detected, self-generating file by inferring package dependencies."
-            % func.__name__
-        )
+        if os.path.exists(REQUIREMENTS_FILE):
```

Comment:
Lets not nest conditionals unnecessarily. This can be:
```
elif os.path.exists(REQUIRMENTS_FILE)
```
(and the else below would be unindented too)